### PR TITLE
Use `mesheryctl` to deploy service meshes

### PR DESCRIPTION
**Pull request by s1ntaxe770r**
Signed-off-by: Jubril Oyetunji <cyberdev01@protonmail.com>

**Description**
I added a bash script to check if mesheryctl is installed and deploy a service mesh
This PR fixes #
#20 
**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/scripts/mesh_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+MESH=$1
+
+echo "checking if mesheryctl is installed"
+	
+if mesheryctl
+	
+then
+	echo found mesheryctl, deploying $MESH
```
Comment (Aisuko) [inline .github/workflows/scripts/mesh_deploy.sh:8]: Is there we need add `"`?

File: .github/workflows/scripts/mesh_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	mesheryctl mesh deploy $mesh
```
Comment (Aisuko) [inline .github/workflows/scripts/mesh_deploy.sh:9]: `$MESH`?

File: .github/workflows/scripts/mesh_deploy.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	
+fi
+
+
+
+
+
+install_mesheryctl(){
+  curl -L https://github.com/meshery/meshery/releases/download/v0.5.67/mesheryctl_0.5.67_Linux_x86_64.zip -o mesheryctl.zip
+  unzip -n mesheryctl.zip
```
Comment (Aisuko) [inline .github/workflows/scripts/mesh_deploy.sh:25]: Is there any reason we shouldn't overwrite the existing files?

File: .github/workflows/scripts/mesh_deploy.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
+MESH=$1
+
+echo "checking if mesheryctl is installed"
```
Comment (Aisuko) [inline .github/workflows/scripts/mesh_deploy.sh:3]: `Cheking` would be good.

File: .github/workflows/scripts/mesh_deploy.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+fi
+
+
+
+
+
+install_mesheryctl(){
+  curl -L https://github.com/meshery/meshery/releases/download/v0.5.67/mesheryctl_0.5.67_Linux_x86_64.zip -o mesheryctl.zip
+  unzip -n mesheryctl.zip 
+  mv mesheryctl /usr/local/bin/mesheryctl
```
Comment (Aisuko) [inline .github/workflows/scripts/mesh_deploy.sh:26]: If I remember right, there would be a permission issue? Maybe there is other solution here?https://unix.stackexchange.com/questions/37724/permissions-ownership-of-usr-local-bin

**Review by Aisuko — COMMENTED** [review]
Thanks for contributing, please take a look the comment below.

File: .github/workflows/scripts/mesh_deploy.sh (hunk 5 above)
Comment (s1ntaxe770r) [reply .github/workflows/scripts/mesh_deploy.sh:26]: 🤔 I'm not sure, as far as I know, workflow runners usually run as a privileged user

Comment (Aisuko) [reply .github/workflows/scripts/mesh_deploy.sh:26]: Sounds good. We don't need to change this.

File: .github/workflows/scripts/mesh_deploy.sh [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	mesheryctl mesh deploy $MESH
+	
+fi
+
+
+
+
+
+install_mesheryctl(){
+  curl -L https://github.com/meshery/meshery/releases/download/v0.5.67/mesheryctl_0.5.67_Linux_x86_64.zip -o mesheryctl.zip
```
Comment (leecalcote) [inline .github/workflows/scripts/mesh_deploy.sh:24]: Don't hardcode the version. Use the same command from the official docs: https://meshery.io. Any reason to deviate?

File: .github/workflows/scripts/mesh_deploy.sh (hunk 5 above)
Comment (leecalcote) [reply .github/workflows/scripts/mesh_deploy.sh:26]: If you use the approved installation procedure, this is already considered for.

// @hexxdump

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by navendu-pottekkat** [comment]
@s1ntaxe770r Could you also add a check in the SMP action to see if Meshery is already installed? Skip install if the environment already has Meshery.

We should also onboard an application on the mesh once it is installed. We could leverage the [mesheryctl-service-mesh-patterns-action](https://github.com/layer5io/mesheryctl-service-mesh-patterns-action) for this I guess or we could use mesheryctl in the bash script directly as well.

// @sayantan1413 @utkarsh-pro 

Once we have that, we can switch the scripts used in the scheduled benchmarks workflow.

**Comment by s1ntaxe770r** [comment]
@navendu-pottekkat is the action functional now? i haven't been following progress on that. 
🤔  about checking if meshery is installed, I thought we were leaving that to the user ?  I could be wrong on this

**Comment by leecalcote** [comment]
@s1ntaxe770r tests run daily - https://github.com/layer5io/meshery-smp-action/actions

**Comment by s1ntaxe770r** [comment]
I'll  address the failing osm deployment in a separate PR

**Comment by navendu-pottekkat** [comment]
@s1ntaxe770r The action is functional now. We should ideally check if Meshery is already deployed in the action as I mentioned above. And we should also include deploying the service mesh and an application on the service mesh in this script.

This script takes in arguments, which is good so we can just plug this script in the workflow to deploy a mesh and an application on top of the mesh.

@Revolyssup Can point you to a pattern file that does this. We can start with one mesh and then make progress.

**Comment by Revolyssup** [comment]
> @s1ntaxe770r The action is functional now. We should ideally check if Meshery is already deployed in the action as I mentioned above. And we should also include deploying the service mesh and an application on the service mesh in this script.
> 
> This script takes in arguments, which is good so we can just plug this script in the workflow to deploy a mesh and an application on top of the mesh.
> 
> @Revolyssup Can point you to a pattern file that does this. We can start with one mesh and then make progress.

To get a little context. a patternfile that does "what"? Installs a service mesh?

**Comment by navendu-pottekkat** [comment]
@Revolyssup We will use `mesheryctl mesh deploy` to deploy the service mesh. We need a pattern file that say, onboards BookInfo onto Istio. We want to use mesheryctl for deploying a mesh and onboarding an application, so a pattern file that can onboard applications is what is needed here.

**Comment by leecalcote** [comment]
Gents, this has gone stale. What are the next steps here?

**Comment by s1ntaxe770r** [comment]
@leecalcote  i think we need to merge this into one review, i accidentally pushed a fix for #25 and #26  over here
https://github.com/layer5io/meshery-smp-action/pull/26

**Comment by leecalcote** [comment]
So close this PR?

**Comment by leecalcote** [comment]
This has become stale again, @s1ntaxe770r @navendu-pottekkat. 

Should we toss in the towel?

**Comment by s1ntaxe770r** [comment]
Nope, a fix is available in #26

**Comment by navendu-pottekkat** [comment]
We can close this and consolidate it to #26

**Comment by leecalcote** [comment]
@gyohuangxin will you review and decide which direction to go from here?

**Comment by l5io** [comment]
PR abandoned.

**Pull request closed without merging**